back/app/controllers/web_api/v1/ideas_controller.rb

```diff
@@ -291,7 +291,7 @@ def similar_ideas
 
     title_threshold = phase_for_input.similarity_threshold_title
     body_threshold = phase_for_input.similarity_threshold_body
-    cache_key = "similar_ideas/#{{ title_multiloc: idea.title_multiloc, body_multiloc: idea.body_multiloc, project_id: idea.project_id, title_threshold:, body_threshold: }}"
+    cache_key = "similar_ideas/#{{ id: idea.id, title_multiloc: idea.title_multiloc, body_multiloc: idea.body_multiloc, project_id: idea.project_id, title_threshold:, body_threshold: }}"
```

Comment:
I wonder if this is a desired change. When different users start writing the same text we will no longer hit the cache. So this will have a negative impact on performance, but it's hard to tell how acceptable this is. I think we need at least a test or comment explaining why this has been added (it might be tempting to remove when tackling performance).

Reply:
@EdwinKato I do have a preference to not include the ID in the cache key, but I suspect we would still need to do filtering. If someone searched for the same title/body on another idea, the cashed result could still contain the current idea I think? It's quite an edge case though.